### PR TITLE
libc: add printf and friends

### PR DESCRIPTION
**Pull request by jcarlosv**
add string functions add printf

Ported from legacy branch

Signed-off-by: Carlos Venegas <vmjcarlos@gmail.com>

### CODE REVIEW

**Review by aduejazz — APPROVED** [review]

File: kernel/kernel.c [hunk 1]
```diff
@@ -1,15 +1,12 @@
 #include <stdint.h>
+#include <stdio.h>
```
Comment (aduejazz) [inline kernel/kernel.c:2]: Could this clash with the machine stdio.h ? shouldn't it be "stdio.h" or "include/stdio.h" ?

Comment (jcarlosv) [reply kernel/kernel.c:2]: I agree, I would go for own header names. `io.h` :P ?  . Same for printf I would go for "info 
, debug, warn, err" names.

Comment (aduejazz) [reply kernel/kernel.c:2]: i guess the name could be ok but the location may clash with the machine where you are building ... afaik <stdio.h> means go to your machine /usr/include to fine the header "stdio.h" means look for it in repo or where you added the "-IXXXX"

Comment (aduejazz) [reply kernel/kernel.c:2]: *find the header

Comment (josuedhg) [reply kernel/kernel.c:2]: it depends you can redefine that in your system, and those bin stuffs just can be used if you compiller is compiled with the flag withsysroot enabled

Comment (josuedhg) [reply kernel/kernel.c:2]: other wise the compiler will try to find in your system

File: kernel/Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@ CC := gcc
 AS := as
 LD := ld
 OBJDUMP := /usr/bin/objdump
-CFLAGS := -g -std=gnu99 -nostdlib -ffreestanding -fno-pie -O0 -Wextra -m32
+CFLAGS := -g -std=gnu11 -nostdlib -ffreestanding -fno-pie -O0 -Wextra -m32 -Ilibc/include
```
Comment (josuedhg) [inline kernel/Makefile:7]: why this change?
